osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/infra/jms/protocol/in/messageprocessors/PushNotificationSmsMessageProcessor.java

```diff
@@ -63,30 +64,34 @@ public void processMessage(final ObjectMessage message) throws JMSException {
 
             if (pushNotificationSms.getIpAddress() != null && !"".equals(pushNotificationSms.getIpAddress())) {
 
-                LOGGER.info("Updating device {} IP address from {} to {}", deviceIdentification,
+                LOGGER.info("Updating device {} IP address from {} to {}", metadata.getDeviceIdentification(),
                         requestMessage.getIpAddress(), pushNotificationSms.getIpAddress());
 
                 // Convert the IP address from String to InetAddress.
                 final InetAddress address = InetAddress.getByName(pushNotificationSms.getIpAddress());
 
                 // Lookup device
-                final Device device = this.deviceRepository.findByDeviceIdentification(deviceIdentification);
+                final Device device = this.deviceRepository.findByDeviceIdentification(
+                        metadata.getDeviceIdentification());
                 if (device != null) {
                     device.updateRegistrationData(address, device.getDeviceType());
+                    device.updateConnectionDetailsToSuccess();
                     this.deviceRepository.save(device);
                 } else {
                     LOGGER.warn(
-                            "Device with ID = {} not found. Discard Sms notification request from ip address = {} of device",
-                            deviceIdentification, address);
+                            "Device with ID = {} not found. Discard Sms notification request from ip address = {} of "
```

Comment:
it is unclear why the alarm processor throws an exception when the Device is not found and this one just logs a warning. Will discuss this with Gerbert

Reply:
I will also throw an exception when device not found.